# Tablas de frecuencias

In [4]:
#CORRER ESTA CELDA PRIMERO ANTES DE EJECUTAR LOS EJERCICIOS PARA QUE NO DEN ERROR
import pandas as pd

## Objetivos:

Aplicar el análisis de frecuencias en distintas variables según su clasificación e interpretar los resultados.

*Nota: Este laboratorio está diseñado para desarrollar razonamiento estadístico. Evita el uso de herramientas de IA para responder las preguntas a menos que esto sea explícitamente sugerido* 🤖. *En su lugar, enfócate en aplicar los conceptos revisados.*

## Actividad 1

Netflix es una de las plataformas de streaming más populares del mundo. En esta actividad, trabajarás con una base de datos real de títulos disponibles en Netflix. El objetivo es explorar algunas características del contenido y observar qué tan comunes son ciertos tipos de programas o películas.

Descarga la base de datos <a href="https://docs.google.com/spreadsheets/d/1E9w_qc7U09eaEcybEw08SOB5-JuIvSlM/export?format=xlsx">aquí</a> y responde las siguientes preguntas:

In [3]:
df_1 = pd.read_excel('Base_Netflix.xlsx')
df_1

ID      Tipo                 Título  \
0       s47  Película             Safe House   
1       s82  Película                   Kate   
2      s136  Película            Cliffhanger   
3      s142  Película             Extraction   
4      s229  Película       The November Man   
...     ...       ...                    ...   
7090  s1066   TV Show                Slasher   
7091  s1702   TV Show  American Horror Story   
7092  s5889   TV Show          Hemlock Grove   
7093  s6918   TV Show               Hannibal   
7094  s5264   TV Show            Dark Matter   

                                       País   Año   Duración  \
0        South Africa, United States, Japan  2012    115 min   
1                             United States  2021    106 min   
2       United States, Italy, France, Japan  1993    113 min   
3     United States, United Kingdom, Canada  2015     82 min   
4             United States, United Kingdom  2014    108 min   
...                                     ...   ...        ...   
7090                                 Canada  2019  3 Seasons   
7091                          United States  2019  9 Seasons   
7092                          United States  2015  3 Seasons   
7093                          United States  2015  3 Seasons   
7094                                 Canada  2017  3 Seasons   

              Categoría  
0     Acción y Aventura  
1     Acción y Aventura  
2     Acción y Aventura  
3     Acción y Aventura  
4     Acción y Aventura  
...                 ...  
7090             Horror  
7091             Horror  
7092             Horror  
7093             Horror  
7094    Ciencia Ficción  

[7095 rows x 7 columns]

1. Defina la población y muestra.

In [4]:
muestra1 = df_1.shape[0] #numero de filas
muestra1

7095

La poblacion es peliculas y la muestra son 7095 peliculas de netflix.

2. ¿Qué clasificación tiene la variable *Tipo*?

Es una variable cualitativa nominal

3. ¿Qué tipo de contenido es más frecuente en la plataforma: *Película* o *TV show* (series)?

In [5]:
conteo = df_1['Tipo'].value_counts()
print(conteo)

Tipo
Película    5673
TV Show     1422
Name: count, dtype: int64


Es mas frecuente las peliculas con 5673 vs 1422 de las series

4. ¿Qué clasificación tiene la variable *Categoría*?

Es variable cualitativa nominal

5. ¿Cuál es la categoría de contenido más común? ¿A qué crees que se debe esto?

In [6]:
conteo2 = df_1['Categoría'].value_counts()
print(conteo2)

Categoría
Dramas                  1593
Comedia                 1266
Acción y Aventura        856
Documental               759
Películas familiares     500
Crimen                   358
Stand-Up Comedy          334
Niños                    318
Horror                   273
Docuseries               175
Anime                    173
Películas                165
Clásicos                 100
Reality Show              99
Thrillers                 62
Romántico                 35
Musical                   15
Ciencia Ficción           14
Name: count, dtype: int64


La categoria mas comun es Dramas con 1593.

6. Crea una tabla que contenga las frecuencias absolutas y relativas de la cantidad de títulos por categoría. Luego, interpreta la frecuencia relativa de las categorías *Documental* y *Comedia*, y responde ¿a qué razón atribuyes la diferencia entre los valores de las frecuencias?

In [7]:
frecuencia_abs = df_1['Categoría'].value_counts()
frecuencia_rel = df_1['Categoría'].value_counts(normalize=True)

tabla = pd.DataFrame({
    'Frecuencia Absoluta': frecuencia_abs,
    'Frecuencia Relativa': frecuencia_rel.round(3)* 100  # redondear a 3 decimales
})

print(tabla)


                      Frecuencia Absoluta  Frecuencia Relativa
Categoría                                                     
Dramas                               1593                 22.5
Comedia                              1266                 17.8
Acción y Aventura                     856                 12.1
Documental                            759                 10.7
Películas familiares                  500                  7.0
Crimen                                358                  5.0
Stand-Up Comedy                       334                  4.7
Niños                                 318                  4.5
Horror                                273                  3.8
Docuseries                            175                  2.5
Anime                                 173                  2.4
Películas                             165                  2.3
Clásicos                              100                  1.4
Reality Show                           99              

La Comedia tiene mayor representación en el catálogo que los Documentales, lo que refleja la preferencia del mercado y de los productores por géneros de entretenimiento masivo, frente a los documentales que cumplen un rol más especializado.

7. Construya una tabla de frecuencias absolutas y relativas de los 10 países con mayor cantidad de títulos, ¿es uniforme la presencia en la plataforma de estos 10 países? ¿Por qué piensas que ocurre lo anterior?

In [8]:
frecuencia_abs2  = df_1['País'].value_counts().head(10)

frecuencia_rel2  = df_1['País'].value_counts(normalize=True).head(10)


tabla2 = pd.DataFrame({
    'Frecuencia Absoluta': frecuencia_abs2,
    'Frecuencia Relativa': frecuencia_rel2.round(3)* 100  # redondear a 3 decimales
})

print(tabla2)


                Frecuencia Absoluta  Frecuencia Relativa
País                                                    
United States                  2779                 39.2
India                           918                 12.9
United Kingdom                  221                  3.1
Japan                           213                  3.0
Canada                          166                  2.3
Spain                           120                  1.7
France                          106                  1.5
Mexico                           96                  1.4
Egypt                            94                  1.3
Nigeria                          87                  1.2


La tabla muestra que Estados Unidos concentra el 39.2% de los títulos disponibles, seguido por India con 12.9%. El resto de países tiene participaciones mucho menores (entre 1% y 3% cada uno).

Esto evidencia que la presencia no es uniforme entre los 10 países, ya que hay una alta concentración en pocos mercados: Estados Unidos y la India acumulan más de la mitad del total.

## Actividad 2

La siguiente base de datos contiene información de estudiantes pertenecientes a un Instituto Profesional. Con el fin de tomar decisiones institucionales, como reforzar asignaturas o rediseñar horarios se registraron datos sobre su rendimiento académico, género, jornada y número de asignaturas reprobadas.

Para responder las siguientes preguntas, descarga la base de datos **<a href="https://drive.google.com/uc?export=download&id=1Z2mrhxQnGgEfMqKrPyXBBPLnFPg52Woz
">aquí</a>**.

In [10]:
df_2 = pd.read_csv('Calificaciones.csv')
df_2

id     Género     Jornada Asignaturas reprobadas Semestre ingreso
0       1   Femenino       Diuna                      0          Primero
1       2   Femenino       Diuna                      3          Primero
2       3  Masculino       Diuna                      4          Primero
3       4  Masculino       Diuna                      1          Segundo
4       5   Femenino       Diuna                      3          Primero
..    ...        ...         ...                    ...              ...
995   996  Masculino       Diuna                      3          Segundo
996   997   Femenino  Vespertina                      3          Primero
997   998   Femenino  Vespertina                      1          Segundo
998   999  Masculino       Diuna                      1          Segundo
999  1000  Masculino  Vespertina                      2          Segundo

[1000 rows x 5 columns]

1. Define la población y la muestra del estudio.

In [11]:
muestra2 = df_2.shape[0] #numero de filas
muestra2

1000

La poblacion son los estudiantes de un instituto tecnico, siendo de 1000 estudiantes la muestra

2. ¿Cuál es la clasificación de las variables presentes en el estudio?

Genero = cualitativa nominal
Jornada = cualitativa nominal
asginaturas reprobadas = cuantitativa discreta
Semestre ingreso = cualitativa nominal

3. ¿Cuál es el género más común entre los estudiantes? ¿Qué porcentaje representa?

In [12]:
conteo = df_2['Género'].value_counts()
frecuencia_rel2  = df_2['Género'].value_counts(normalize=True).head(10) * 100
print(conteo)
print(frecuencia_rel2)

Género
Femenino     505
Masculino    495
Name: count, dtype: int64
Género
Femenino     50.5
Masculino    49.5
Name: proportion, dtype: float64


El Genero mas comun es el masculino con un 50.5%.

4. ¿Qué jornada tiene mayor representación en la muestra? ¿Hay alguna diferencia relevante entre el porcentaje de estudiantes que pertenece a cada una?

In [13]:
conteoJornada = df_2['Jornada'].value_counts()
frecuencia_relJor  = df_2['Jornada'].value_counts(normalize=True).head(10) * 100
print(conteoJornada)
print(frecuencia_relJor)

Jornada
Diuna         649
Vespertina    351
Name: count, dtype: int64
Jornada
Diuna         64.9
Vespertina    35.1
Name: proportion, dtype: float64


La joranda diurna tiene 64.9% de representacion de la muestra, hay una diferencia del 29.8% respecto a los vespertinos

5. Realiza una tabla de frecuencias para la variable Asignaturas reprobadas. La tabla debe mostrar las frecuencias puntuales y acumuladas, luego responde las siguientes preguntas:

In [14]:
# Tomar la columna correcta
asignaturas = df_2['Asignaturas reprobadas']  # Usando exactamente el nombre de la columna
tamaño = len(asignaturas)

# Frecuencias
frecuencias = asignaturas.value_counts().sort_index()

# Tabla de frecuencias
tabla_frecuencias_asignaturas = pd.DataFrame({
    'Frec. absoluta fi': frecuencias,
    'Frec. relativa hi (%)': (frecuencias / tamaño) * 100,
    'Frec. abs. acumulada Fi': frecuencias.cumsum(),
    'Frec. rel. acumulada Hi(%)': (frecuencias.cumsum() / tamaño) * 100
})

# Redondear a 1 decimal
tabla_frecuencias_asignaturas = round(tabla_frecuencias_asignaturas, 1)

tabla_frecuencias_asignaturas



Frec. absoluta fi  Frec. relativa hi (%)  \
Asignaturas reprobadas                                             
0                                      81                    8.1   
1                                     200                   20.0   
2                                     211                   21.1   
3                                     199                   19.9   
4                                     209                   20.9   
5 o más                               100                   10.0   

                        Frec. abs. acumulada Fi  Frec. rel. acumulada Hi(%)  
Asignaturas reprobadas                                                       
0                                            81                         8.1  
1                                           281                        28.1  
2                                           492                        49.2  
3                                           691                        69.1  
4                                           900                        90.0  
5 o más                                    1000                       100.0

  $\hspace{1cm}$ 5.1 ¿Cuál es la cantidad más común de asignaturas reprobadas?

La cantidad mas comun reprobadas son 2, con un 21.1% de F.relativa

$\hspace{1cm}$ 5.2. ¿Qué porcentaje de estudiantes ha reprobado 3 asignaturas o menos?

El 69.1% de los estudiantes han reprobado entre 3 y 0 asignaturas.

$\hspace{1cm}$ 5.3. Interpreta las siguientes frecuencias:

$\hspace{1cm}$ $F_{3}$: 492 estudiantes han reprobado entre 2 y 0 asignaturas

$\hspace{1cm}$ $f_{1}$: 81 estudiantes han reprobado 0 asignaturas

$\hspace{1cm}$ $H_{5}$: EL 90% de los estudiantes han reprobado hasta 4 asignaturas

$\hspace{1cm}$ $h_{2}$: El 20% de los estudiantes han reprobado 1 asignatura.

## Actividad 3

La base de datos utilizada en esta actividad corresponde al Informe Mundial de la Felicidad publicado por la ONU del año 2024. Contiene información de 140 países y evalúa el nivel de felicidad de sus habitantes en una escala continua (de 0 a 10), así como otros indicadores como el PIB per cápita, apoyo social, esperanza de vida saludable, percepción de corrupción y libertad para tomar decisiones de vida.

Para responder las siguientes preguntas, descarga la base de datos **<a href="https://drive.google.com/uc?export=download&id=1bKzL1MFdbfXMy9aYpGNt-SXlTrxEEL_9
">aquí</a>**.

*Log PIB per cápita, corresponde al logaritmo del Producto Interno Bruto per cápita. Esta transformación se usa para reducir la asimetría de la distribución del PIB per cápita y facilitar comparaciones cuando hay grandes diferencias entre países.*


In [15]:
df_3 = pd.read_csv('Reporte_felicidad_2024.csv')
df_3

Nombre del país               Indicador regional  Puntaje de felicidad  \
0           Finlandia                Europa occidental                 7.741   
1           Dinamarca                Europa occidental                 7.583   
2            Islandia                Europa occidental                 7.525   
3              Suecia                Europa occidental                 7.344   
4              Israel  Medio Oriente y Norte de África                 7.341   
..                ...                              ...                   ...   
135  Congo (Kinshasa)              África subsahariana                 3.295   
136      Sierra Leona              África subsahariana                 3.245   
137            Lesoto              África subsahariana                 3.186   
138            Líbano  Medio Oriente y Norte de África                 2.707   
139        Afganistán                     Asia del Sur                 1.721   

     Log PIB per cápita  Apoyo social  Esperanza de vida saludable  \
0                 1.844         1.572                        0.695   
1                 1.908         1.520                        0.699   
2                 1.881         1.617                        0.718   
3                 1.878         1.501                        0.724   
4                 1.803         1.513                        0.740   
..                  ...           ...                          ...   
135               0.534         0.665                        0.262   
136               0.654         0.566                        0.253   
137               0.771         0.851                        0.000   
138               1.377         0.577                        0.556   
139               0.628         0.000                        0.242   

     Libertad para tomar decisiones de vida,  Generosidad  \
0                                      0.859        0.142   
1                                      0.823        0.204   
2                                      0.819        0.258   
3                                      0.838        0.221   
4                                      0.641        0.153   
..                                       ...          ...   
135                                    0.473        0.189   
136                                    0.469        0.181   
137                                    0.523        0.082   
138                                    0.173        0.068   
139                                    0.000        0.091   

     Percepciones de corrupción  Distopía + residual  
0                         0.546                2.082  
1                         0.548                1.881  
2                         0.182                2.050  
3                         0.524                1.658  
4                         0.193                2.298  
..                          ...                  ...  
135                       0.072                1.102  
136                       0.053                1.068  
137                       0.085                0.875  
138                       0.029               -0.073  
139                       0.088                0.672  

[140 rows x 10 columns]

1. Define la población y muestra del estudio.

In [17]:
muestra3 = df_3.shape[0] #numero de filas
muestra3

140

La poblacion serian los paises y la muestra serian 140 paises.

2. ¿Cuál es la clasificación de la variable *Puntaje de felicidad*?

Cuantitativa Discreta

3. Crea una tabla de frecuencias puntuales y acumuladas utilizando la variable *Puntaje de felicidad*. Considera crear la tabla de frecuencias con 8 intervalos.

In [18]:
df_3 = pd.read_csv('Reporte_felicidad_2024.csv')
intervalos = 8
# Vamos a agregar una nueva columna, con la función cut()
df_3['Intervalos'] = pd.cut(df_3['Puntaje de felicidad'], bins = intervalos, include_lowest=True)
# Frecuencias
f = df_3.groupby('Intervalos').size()
F = f.cumsum()
h = ((f/140)*100).round(1)
H = h.cumsum()
# Armar el DataFrame
tabla_frecuencias_ejemplo_3 = pd.DataFrame({
    'Frecuencia absoluta fi': f,
    'Frecuencia relativa hi (%)': h,
    'Frecuencia absoluta acumulada Fi': F,
    'Frecuencia relativa acumulada Hi (%)': H})
tabla_frecuencias_ejemplo_3

/var/folders/rn/64_qx9bd1f763wmgvnt9chfc0000gn/T/ipykernel_88785/2818214775.py:6: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  f = df_3.groupby('Intervalos').size()


Frecuencia absoluta fi  \
Intervalos                                            
(1.7140000000000002, 2.474]                       1   
(2.474, 3.226]                                    2   
(3.226, 3.978]                                   14   
(3.978, 4.731]                                   19   
(4.731, 5.483]                                   25   
(5.483, 6.236]                                   34   
(6.236, 6.988]                                   34   
(6.988, 7.741]                                   11   

                             Frecuencia relativa hi (%)  \
Intervalos                                                
(1.7140000000000002, 2.474]                         0.7   
(2.474, 3.226]                                      1.4   
(3.226, 3.978]                                     10.0   
(3.978, 4.731]                                     13.6   
(4.731, 5.483]                                     17.9   
(5.483, 6.236]                                     24.3   
(6.236, 6.988]                                     24.3   
(6.988, 7.741]                                      7.9   

                             Frecuencia absoluta acumulada Fi  \
Intervalos                                                      
(1.7140000000000002, 2.474]                                 1   
(2.474, 3.226]                                              3   
(3.226, 3.978]                                             17   
(3.978, 4.731]                                             36   
(4.731, 5.483]                                             61   
(5.483, 6.236]                                             95   
(6.236, 6.988]                                            129   
(6.988, 7.741]                                            140   

                             Frecuencia relativa acumulada Hi (%)  
Intervalos                                                         
(1.7140000000000002, 2.474]                                   0.7  
(2.474, 3.226]                                                2.1  
(3.226, 3.978]                                               12.1  
(3.978, 4.731]                                               25.7  
(4.731, 5.483]                                               43.6  
(5.483, 6.236]                                               67.9  
(6.236, 6.988]                                               92.2  
(6.988, 7.741]                                              100.1

4. Interpreta las siguientes frecuencias:

$\hspace{1cm}$ $f_{3}$: Hay 14 paises que tienen un puntaje de felicidad entre el 3226 y 3978

$\hspace{1cm}$ $F_{5}$: 61 paises tienen un puntaje de felicidad de entre el 1714 al 5483

$\hspace{1cm}$ $h_{4}$:Hay 13.6% de los paises que tienen entre 3978 a 4731 puntos de felicidad

$\hspace{1cm}$ $H_{6}$: El 67.9% de los paises tiene entre 1714 y 6236 puntos de felicidad

5. ¿Qué puedes concluir sobre la concentración de países según su nivel de felicidad? Para responder considera las siguientes categorías de nivel de felicidad:

* Baja: (1,70; 3,23]
* Media: (3,23; 4,73]
* Media-alta: (4,73; 6,24]
* Alta: (6,24; 7,74]

Hay un 3% porciento de los paises que tienen un indice bajo de felicidad.

Hay un 23.6% d elos paises que se situa en el nivel medio de felicidad.

Hay un 42,2% de los paises que se situa en felicidad media-alta.

Hay un 32,2% de los paises que se situa en felicidad alta.

Esto quiere decir que hay mas paises que sienten felicidad media-alta que el resto de las clasificaciones.

6. ¿Cómo se distribuyen los países según su PIB per cápita? Construye una tabla de frecuencias considerando 8 intervalos.

In [20]:
df_3 = pd.read_csv('Reporte_felicidad_2024.csv')
intervalos = 8
# Vamos a agregar una nueva columna, con la función cut()
df_3['Intervalos'] = pd.cut(df_3['Log PIB per cápita'], bins = intervalos, include_lowest=True)
# Frecuencias
f = df_3.groupby('Intervalos').size()
F = f.cumsum()
h = ((f/140)*100).round(1)
H = h.cumsum()
# Armar el DataFrame
tabla_frecuencias_ejemplo_4 = pd.DataFrame({
    'Frecuencia absoluta fi': f,
    'Frecuencia relativa hi (%)': h,
    'Frecuencia absoluta acumulada Fi': F,
    'Frecuencia relativa acumulada Hi (%)': H})
tabla_frecuencias_ejemplo_4

/var/folders/rn/64_qx9bd1f763wmgvnt9chfc0000gn/T/ipykernel_88785/1726507728.py:6: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  f = df_3.groupby('Intervalos').size()


Frecuencia absoluta fi  Frecuencia relativa hi (%)  \
Intervalos                                                              
(-0.00314, 0.268]                       1                         0.7   
(0.268, 0.535]                          1                         0.7   
(0.535, 0.803]                         17                        12.1   
(0.803, 1.07]                          15                        10.7   
(1.07, 1.338]                          23                        16.4   
(1.338, 1.606]                         33                        23.6   
(1.606, 1.873]                         37                        26.4   
(1.873, 2.141]                         13                         9.3   

                   Frecuencia absoluta acumulada Fi  \
Intervalos                                            
(-0.00314, 0.268]                                 1   
(0.268, 0.535]                                    2   
(0.535, 0.803]                                   19   
(0.803, 1.07]                                    34   
(1.07, 1.338]                                    57   
(1.338, 1.606]                                   90   
(1.606, 1.873]                                  127   
(1.873, 2.141]                                  140   

                   Frecuencia relativa acumulada Hi (%)  
Intervalos                                               
(-0.00314, 0.268]                                   0.7  
(0.268, 0.535]                                      1.4  
(0.535, 0.803]                                     13.5  
(0.803, 1.07]                                      24.2  
(1.07, 1.338]                                      40.6  
(1.338, 1.606]                                     64.2  
(1.606, 1.873]                                     90.6  
(1.873, 2.141]                                     99.9

7. Considera las siguientes categorías de nivel de PIB per cápita:
* Baja: (-0,0121; 0,54]
* Media: (0,54; 1,07]
* Media-alta: (1,07; 1,61]
* Alta: (1,61; 2,14]

Con base en la tabla de frecuencias del inciso anterior, ¿en qué categoría se concentra la mayoría de los países y qué conclusiones puedes extraer sobre su desarrollo económico?

La mayoria de los paises se concentran en la categoria Alta, indicando así que son mayoritariamente paises con mas PIB.

8. En el siguiente gráfico se representa la relación entre el logaritmo del PIB per cápita y el puntaje de felicidad, donde cada punto corresponde a un país. ¿Qué ocurre con los países que tienen menor valor de logaritmo del PIB per capita? ¿Y con los que tienen un mayor valor? ¿Observas alguna tendencia?

<img src = "https://i.ibb.co/WNw0PtRf/Captura-de-pantalla-2025-07-22-102447.png" width = "800"/>

Hay una corelacion directa entre la felicidad y el pib, indicando asi que los paises con mayor pib tienen tambien mayores indices de felicidad

## Actividad 4

Para describir un conjunto de llamadas telefónicas se trabajará con una base de datos que contiene información del tiempo de duración y destino de 250  llamadas realizadas a algún país de América del Norte.

Descarga la base de datos **<a href="https://docs.google.com/spreadsheets/d/1KtiWhqgBecg_sKsGur_OCog5a0sU69NQ/export?format=xlsx
">aquí</a>**.


1. Define la población y la muestra del estudio.

La Población es el conjunto total de todas las llamadas telefónicas realizadas a países de América del Norte en el contexto de estudio, incluyendo las llamadas no registradas. Mientras que la Muestra son las 250 llamadas específicas registradas en la base de datos, con detalles sobre su duración y destino.

2. Muestra las tres primeras filas y tres últimas filas de DataFrame.

In [6]:
df_4 = pd.read_excel('Llamadas.xlsx')
df_4.head(3)

Registro llamada         Destino  Tiempo (en minutos)
0         150131774          Canadá                 35.3
1         150131698          Canadá                 19.4
2         150131725  Estados Unidos                 50.4

In [10]:
df_4.tail(3)

Registro llamada         Destino  Tiempo (en minutos)
247         150131799          México                 39.1
248         150131663          México                 20.3
249         150131624  Estados Unidos                 34.8

3. Clasifica las variables relevantes en el estudio.

Registro llamada: Cualitativa nominal (es un identificador único, como un código o ID, sin orden ni magnitud).

Destino: Cualitativa nominal (categorías: Canadá, Estados Unidos, México; no hay orden inherente entre ellas).

Tiempo (en minutos): Cuantitativa continua (mide duración en minutos, puede tomar cualquier valor decimal en un rango, como 35.3 o 50.4).

4. Construye una tabla de distribución de frecuencias para la variable tiempo de duración de las llamadas considerando frecuencias puntuales y acumuladas. Para la construcción de la tabla considera 9 intervalos.

In [ ]:
df_4 = pd.read_excel('Llamadas.xlsx')

intervalos = 9
df_4['Intervalos'] = pd.cut(df_4['Tiempo (en minutos)'], bins=intervalos, include_lowest=True)

# Frecuencias
f = df_4.groupby('Intervalos').size()
F = f.cumsum()
h = ((f / 250) * 100).round(1)  # Total de observaciones es 250, en porcentaje, redondeado a 1 decimal
H = h.cumsum()

# Armar el DataFrame
tabla_frecuencias_ejemplo_4 = pd.DataFrame({
    'Frecuencia absoluta fi': f,
    'Frecuencia relativa hi (%)': h,
    'Frecuencia absoluta acumulada Fi': F,
    'Frecuencia relativa acumulada Hi (%)': H
})

# Mostrar la tabla
tabla_frecuencias_ejemplo_4

/var/folders/rn/64_qx9bd1f763wmgvnt9chfc0000gn/T/ipykernel_89235/3520974753.py:9: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  f = df_4.groupby('Intervalos').size()


Frecuencia absoluta fi  \
Intervalos                                           
(4.146999999999999, 9.956]                      12   
(9.956, 15.711]                                 17   
(15.711, 21.467]                                32   
(21.467, 27.222]                                57   
(27.222, 32.978]                                56   
(32.978, 38.733]                                34   
(38.733, 44.489]                                25   
(44.489, 50.244]                                10   
(50.244, 56.0]                                   7   

                            Frecuencia relativa hi (%)  \
Intervalos                                               
(4.146999999999999, 9.956]                         4.8   
(9.956, 15.711]                                    6.8   
(15.711, 21.467]                                  12.8   
(21.467, 27.222]                                  22.8   
(27.222, 32.978]                                  22.4   
(32.978, 38.733]                                  13.6   
(38.733, 44.489]                                  10.0   
(44.489, 50.244]                                   4.0   
(50.244, 56.0]                                     2.8   

                            Frecuencia absoluta acumulada Fi  \
Intervalos                                                     
(4.146999999999999, 9.956]                                12   
(9.956, 15.711]                                           29   
(15.711, 21.467]                                          61   
(21.467, 27.222]                                         118   
(27.222, 32.978]                                         174   
(32.978, 38.733]                                         208   
(38.733, 44.489]                                         233   
(44.489, 50.244]                                         243   
(50.244, 56.0]                                           250   

                            Frecuencia relativa acumulada Hi (%)  
Intervalos                                                        
(4.146999999999999, 9.956]                                   4.8  
(9.956, 15.711]                                             11.6  
(15.711, 21.467]                                            24.4  
(21.467, 27.222]                                            47.2  
(27.222, 32.978]                                            69.6  
(32.978, 38.733]                                            83.2  
(38.733, 44.489]                                            93.2  
(44.489, 50.244]                                            97.2  
(50.244, 56.0]                                             100.0

5. A partir de la pregunta anterior, interpreta las siguientes frecuencias:

$\hspace{1cm}$ $f_{6}$: Se hicieron 34 llamadas que duraron entre 27.2 a 38.7 minutos.

$\hspace{1cm}$ $h_{2}$: El 6.8% de las llamadas registradas duraron entre 9.9 y 15.7 minutos.

$\hspace{1cm}$ $F_{4}$: Un total de 118 llamadas duraron como máximo 27.2 minutos o menos.

$\hspace{1cm}$ $H_{3}$: El 24.4% del total de las llamadas registradas duraron como máximo 21.4 minutos.

## Actividad 5

A continuación, se presentan datos nacionales por comuna de la medición SIMCE de octavo básico rendida en el año 2015.

Descarga la base de datos <a href="https://docs.google.com/spreadsheets/d/1fIGstVw-dQfA5WMq9mvhx3wVOHCXbrpj/export?format=xlsx">aquí</a> y responde las siguientes preguntas:

In [19]:
df_5 = pd.read_excel('SIMCE.xlsx')
df_5

agno  fecha_bbdd      codigo_bbdd grado  cod_reg  \
0    2015    20160908  final20160908v1    8b       11   
1    2015    20160908  final20160908v1    8b        5   
2    2015    20160908  final20160908v1    8b       13   
3    2015    20160908  final20160908v1    8b        8   
4    2015    20160908  final20160908v1    8b        3   
..    ...         ...              ...   ...      ...   
339  2015    20160908  final20160908v1    8b        8   
340  2015    20160908  final20160908v1    8b        8   
341  2015    20160908  final20160908v1    8b        5   
342  2015    20160908  final20160908v1    8b        8   
343  2015    20160908  final20160908v1    8b       13   

                                       nom_reg  cod_pro      nom_pro  cod_com  \
0    AYSÉN DEL GENERAL CARLOS IBÁÑEZ DEL CAMPO        2        AISÉN        1   
1                                   VALPARAÍSO        6  SAN ANTONIO       13   
2                    METROPOLITANA DE SANTIAGO        5    MELIPILLA       43   
3                                       BIOBÍO        3       BIOBÍO        8   
4                                      ATACAMA        3       HUASCO        6   
..                                         ...      ...          ...      ...   
339                                     BIOBÍO        3       BIOBÍO       21   
340                                     BIOBÍO        4        ÑUBLE       53   
341                                 VALPARAÍSO        4      PETORCA       12   
342                                     BIOBÍO        4        ÑUBLE       54   
343                  METROPOLITANA DE SANTIAGO        1     SANTIAGO       25   

             nom_com  ...  nalu_nat8b_com prom_lect8b_com  prom_mate8b_com  \
0              AISÉN  ...             312             237              257   
1          ALGARROBO  ...             146             235              251   
2              ALHUÉ  ...              67             213              236   
3        ALTO BIOBÍO  ...              67             207              200   
4    ALTO DEL CARMEN  ...              50             226              227   
..               ...  ...             ...             ...              ...   
339           YUMBEL  ...             235             241              256   
340           YUNGAY  ...             231             257              271   
341         ZAPALLAR  ...             103             245              267   
342           ÑIQUÉN  ...             105             237              249   
343            ÑUÑOA  ...            2174             259              290   

     prom_nat8b_com  dif_lect8b_com  dif_mate8b_com  dif_nat8b_com  \
0               263             3.0             7.0           -9.0   
1               257            -1.0             2.0          -13.0   
2               240            25.0            15.0          -14.0   
3               218             2.0            -4.0           -9.0   
4               244             8.0             5.0           -2.0   
..              ...             ...             ...            ...   
339             262            11.0            13.0           -3.0   
340             273            13.0            10.0           -9.0   
341             271            -3.0            -4.0           -2.0   
342             250            11.0             9.0           -4.0   
343             291            -1.0             2.0           -7.0   

            sigdif_lect8b_com         sigdif_mate8b_com  \
0            No significativa          No significativa   
1            No significativa          No significativa   
2    Positiva y significativa  Positiva y significativa   
3            No significativa          No significativa   
4            No significativa          No significativa   
..                        ...                       ...   
339  Positiva y significativa  Positiva y significativa   
340  Positiva y significativa  Positiva y significativa   
341          No significativa         

1. Construye una tabla de distribución de frecuencias (absolutas y porcentuales incluyendo las acumuladas) que resuma los puntajes obtenidos en la prueba SIMCE de Matemática. Considera 9 intervalos para la construcción de la tabla.

$\hspace{1cm}$*Observaciones: Se debe escoger la variable prom_mate8b_com*

In [21]:
import pandas as pd

# Definir intervalos manualmente
bins = [200, 213, 226, 239, 252, 265, 278, 291, 304, 317]
df_5['Intervalos'] = pd.cut(df_5['prom_mate8b_com'], bins=bins, include_lowest=True, 
                           labels=[f'[{bins[i]}, {bins[i+1]})' for i in range(len(bins)-1)])

# Calcular frecuencias
f = df_5.groupby('Intervalos').size()
F = f.cumsum()
h = ((f / 344) * 100).round(1)  # Total de comunas es 344, porcentaje redondeado a 1 decimal
H = h.cumsum()

# Crear la tabla de frecuencias
tabla_frecuencias = pd.DataFrame({
    'Frecuencia absoluta (f_i)': f,
    'Frecuencia relativa (h_i %)': h,
    'Frecuencia absoluta acumulada (F_i)': F,
    'Frecuencia relativa acumulada (H_i %)': H
})

# Mostrar la tabla
tabla_frecuencias

/var/folders/rn/64_qx9bd1f763wmgvnt9chfc0000gn/T/ipykernel_89235/2719328562.py:9: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  f = df_5.groupby('Intervalos').size()


Frecuencia absoluta (f_i)  Frecuencia relativa (h_i %)  \
Intervalos                                                           
[200, 213)                          2                          0.6   
[213, 226)                         12                          3.5   
[226, 239)                         54                         15.7   
[239, 252)                        112                         32.6   
[252, 265)                        107                         31.1   
[265, 278)                         45                         13.1   
[278, 291)                          5                          1.5   
[291, 304)                          3                          0.9   
[304, 317)                          3                          0.9   

            Frecuencia absoluta acumulada (F_i)  \
Intervalos                                        
[200, 213)                                    2   
[213, 226)                                   14   
[226, 239)                                   68   
[239, 252)                                  180   
[252, 265)                                  287   
[265, 278)                                  332   
[278, 291)                                  337   
[291, 304)                                  340   
[304, 317)                                  343   

            Frecuencia relativa acumulada (H_i %)  
Intervalos                                         
[200, 213)                                    0.6  
[213, 226)                                    4.1  
[226, 239)                                   19.8  
[239, 252)                                   52.4  
[252, 265)                                   83.5  
[265, 278)                                   96.6  
[278, 291)                                   98.1  
[291, 304)                                   99.0  
[304, 317)                                   99.9

2. ¿Es correcto afirmar que los resultados de Matemática se concentran en los puntajes más bajos? Justifica tu respuesta.

Concentración: La mayoría de las comunas (68% acumulativo hasta [265, 278)) tienen puntajes entre 200 y 278, con un pico en [252, 265) y [265, 278) que suman cerca del 49.4% (23.8% + 25.6%). Los intervalos más bajos ([200, 213) y [213, 226)) representan solo 2.1% acumulado.

Justificación: No es correcto afirmar que los resultados se concentran en los puntajes más bajos (por ejemplo, cerca de 200). La distribución muestra una concentración mayor en el rango medio (252 a 278), con una distribución sesgada hacia valores intermedios y altos. Solo un 7% de las comunas tienen puntajes por debajo de 239, mientras que más del 40% superan los 265, indicando que los puntajes más altos y medios son más representativos.

3.  Diseñe una tabla de frecuencias relativas que resuma la información de la cantidad de comunas por región. Realice observaciones de los datos de la tabla y construya una hipótesis al respecto.

In [25]:
# Contar comunas por región
frecuencia_reg = df_5['nom_reg'].value_counts()
total_comunas = len(df_5)  # 344 comunas
h_reg = ((frecuencia_reg / total_comunas) * 100).round(1)  # Frecuencia relativa en %

# Crear la tabla de frecuencias relativas
tabla_frecuencias_regiones = pd.DataFrame({
    'Frecuencia relativa (h_i %)': h_reg
})

# Mostrar la tabla
tabla_frecuencias_regiones

Frecuencia relativa (h_i %)
nom_reg                                                               
BIOBÍO                                                            15.7
METROPOLITANA DE SANTIAGO                                         15.1
VALPARAÍSO                                                        11.0
LIBERTADOR GENERAL BERNARDO O'HIGGINS                              9.6
LA ARAUCANÍA                                                       9.3
LOS LAGOS                                                          8.7
MAULE                                                              8.7
COQUIMBO                                                           4.4
LOS RÍOS                                                           3.5
AYSÉN DEL GENERAL CARLOS IBÁÑEZ DEL CAMPO                          2.9
ATACAMA                                                            2.6
ANTOFAGASTA                                                        2.6
MAGALLANES Y DE LA ANTÁRTICA CHILENA                               2.6
TARAPACÁ                                                           2.0
ARICA Y PARINACOTA                                                 1.2